src/Simplex/Messaging/Agent.hs

```diff
@@ -384,9 +390,11 @@ createUser' c srvs = do
   atomically $ TM.insert userId srvs $ smpServers c
   pure userId
 
-deleteUser' :: AgentMonad m => AgentClient -> UserId -> m ()
-deleteUser' c userId = do
-  withStore c (`deleteUserRecord` userId)
+deleteUser' :: AgentMonad m => AgentClient -> UserId -> Bool -> m ()
+deleteUser' c userId delSMPQueues = do
+  if delSMPQueues
+    then withStore c (`setUserDeleted` userId) >>= deleteConnectionsAsync' c
```

Comment:
if I understand correctly user record will not be deleted until picked up by cleanup manager, even in case connections are successfully deleted first try. Maybe we could parameterize deleteConnectionsAsync' with "deleteUser" flag and run query that deletes user if no connections exist (in the end of deleteConnQueues?)